packages/flutter/lib/src/painting/continuous_stadium_border.dart

```diff
@@ -0,0 +1,348 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// A stadium border with continuous corners.
+///
+/// A shape similar to a stadium, but with a smoother transition from
+/// each linear edge to its 180º curves. Each 180º curve is approximately half
+/// an ellipse.
+///
+/// In this shape, the curvature of each 180º curve around the length of the arc
+/// is approximately a gaussian curve instead of a step function as with a
+/// traditional half circle round.
+///
+/// In an attempt to keep the shape of the rectangle the same regardless of its
+/// dimension (and to avoid clipping of the shape), the radius will
+/// automatically be lessened if its width or height is less than ~3x the
```

Comment:
Maybe vary the doc here a bit. 'The radius will automatically be lessened'. Lessened from what? There is no user specified radius parameter.

Reply:
The applied radius? What do you mean their is no user specified radius parameter?